### DISCUSSION TITLE
v0.0.1-rewrite.1

### DISCUSSION THREAD

**Discussion by mojidev-py** [mojidev-py/discmoji#49] [Announcements]
Rewrite

**Full Changelog**: https://github.com/mojidev-py/discmoji/compare/patch3...rewrite

<hr /><em>This discussion was created from the release <a href='https://github.com/mojidev-py/discmoji/releases/tag/rewrite'>v0.0.1-rewrite.1</a>.</em>

**Comment by ghost** [comment]
I think I found a bug in one of your recent commits, specifically:

```py
if data is str: ...
elif data is dict: ...
```

Both of these return ` False ` because they are not direct reference to the type class. ` is ` keyword is used to check if a variable is a *direct reference* to something, as oppose to ` == ` equality operator that checks for *value* instead. Take an example below:

```py
a = [123]
b = a
c = [123]

>>> b == a
True

>>> c == a
True

>>> b is a
True

>>> c is a
False
```

Both ` b == a ` and ` c == a ` because all three of them have a value of ` [123] `. However, ` c is a ` is ` False ` because ` 123 ` in the ` c ` list has a different memory pointer ( ? ) than the ` 123 ` in ` a ` list, while ` b is a ` is ` True ` because ` b ` is just a reference to ` a `, you are not declaring another ` [123] `.

If you want to check whether something is an instance or a subclass of something else, use ` isinstance() ` and ` issubclass() ` built-in functions respectively.

```py
test = "hello world"
number = 123

>>> isinstance(test, str)
True

>>> isinstance(number, (int, float))
True
```

**References**
File in question: ` discmoji/types.py `

**Comment by mojidev-py** [comment]
just changed that
